app/components/app-nav-primary.hbs

```diff
@@ -0,0 +1,42 @@
+{{#if this.show}}
+  <div
+    {{focus-trap
+      isActive=(media 'isMobile')
+      focusTrapOptions=(hash
+        onDeactivate=(action @close)
+      )
+    }}>
+    <nav
+      class="app-nav-primary {{if @navPrimaryIsOpen 'app-nav-primary--open'}}"
+      {{click-elsewhere @close isEnabled=this.show}}>
+      <button
+        class="app-nav-primary__mobile-nav-toggle"
+        aria-label="Hide Site Navigation"
+        {{on "click" @close}}>
+        <svg role="image" aria-hidden="true" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" stroke="currentColor">
+          <g>
+            <line x1="3" y1="12" x2="21" y2="12"></line>
+            <line x1="3" y1="6" x2="21" y2="6"></line>
+            <line x1="3" y1="18" x2="21" y2="18"></line>
+          </g>
+        </svg>
+      </button>
+
+      <em>TODO: Make into real links...</em>
+
+      <ul class="app-nav-primary__list">
+        {{#unless isLoading}}
+          {{#each this.contents as | content |}}
+            <li> {{!-- Convert to LinkTo's when we have content page routing done --}}
+              {{content.title}}
+            </li>
+          {{/each}}
+        {{/unless}}
+      </ul>
+    </nav>
+
+    {{#if @navPrimaryIsOpen}}
+      <div class="app-nav-primary-open-bgfade"></div>
```

Comment:
This works, but it still feels a little off to have an empty `<div>` for this. You could, if you wanted, conditionally toggle a class on the `<header>` , `<div class="main-content">`, `<aside>` and `<footer>` that applies a CSS `filter` that achieves the same effect.

Reply:
That'd be nice, or even an `::before` on the nav that does it too. Whichever way you solve it is great, good point!